addons/inter_company_rules/models/purchase_order.py

```diff
@@ -83,7 +83,7 @@ def _prepare_sale_order_data(self, name, partner, company, direct_delivery_addre
             'pricelist_id': partner.property_product_pricelist.id,
             'partner_invoice_id': partner_addr['invoice'],
             'date_order': self.date_order,
-            'fiscal_position_id': partner.property_account_position and partner.property_account_position.id or False,
+            'fiscal_position_id': partner.property_account_position_id and partner.property_account_position_id.id or False,
```

Comment:
Just write: partner.property_account_position_id.id
Accessing a relational field always returns a recordset, empty if the field is not set.